service/src/test/java/uk/gov/hmcts/reform/fpl/utils/NotifyAdminOrderIssuedTestHelper.java

```diff
@@ -0,0 +1,119 @@
+
+        Map<String, Object> resultData = new HashMap<>(dataCaptor.getValue());
+        Map<String, Object> expectedParameters = issuedOrderType == CMO
+            ? getExpectedCMOParametersForAdminWhenRepresentativesNeedServingByPost() :
+            getExpectedParametersForAdminWhenRepresentativesNeedServingByPost();
+
+        assertThat(((JSONObject) resultData.get("caseUrlOrDocumentLink")).get("file")).isEqualTo(
+            ((JSONObject) expectedParameters.get("caseUrlOrDocumentLink")).get("file"));
+
+        resultData.remove("caseUrlOrDocumentLink");
+        expectedParameters.remove("caseUrlOrDocumentLink");
+
+        return Maps.difference(expectedParameters, resultData);
+    }
+
+    public static Map<String, Object> getExpectedParametersForAdminWhenNoRepresentativesServedByPost() {
+        return ImmutableMap.<String, Object>builder()
+            .put("cmoCallout", "")
+            .putAll(commonParametersNoPostingNeeded())
+            .build();
+    }
+
+    public static Map<String, Object> getExpectedCMOParametersForAdminWhenNoRepresentativesServedByPost() {
+        return ImmutableMap.<String, Object>builder()
+            .put("cmoCallout", "^Jones, SACCCCCCCC5676576567")
+            .putAll(commonParametersNoPostingNeeded())
+            .build();
+    }
+
+    public static List<Element<Representative>> buildRepresentativesServedByPost() {
+        return wrapElements(Representative.builder()
+            .email("paul@example.com")
+            .fullName("Paul Blart")
+            .address(Address.builder()
+                .addressLine1("Street")
+                .postTown("Town")
+                .postcode("Postcode")
+                .build())
+            .servingPreferences(POST)
+            .build());
+    }
+
+    private static Map<String, Object> commonParametersNoPostingNeeded() {
+        return ImmutableMap.<String, Object>builder()
+            .put("needsPosting", "No")
+            .put("doesNotNeedPosting", "Yes")
+            .put("courtName", LOCAL_AUTHORITY_NAME)
+            .put("caseUrlOrDocumentLink", formatCaseUrl("http://fake-url", 12345L))
+            .put("respondentLastName", "Jones")
+            .put("representatives", "")
+            .build();
+    }
+
+    private static Map<String, Object> commonParametersPostingNeeded() {
+        String fileContent = new String(Base64.encodeBase64(PDF), ISO_8859_1);
+        JSONObject jsonFileObject = new JSONObject().put("file", fileContent);
+
+        Map<String, Object> expectedMap = new HashMap<>();
+        expectedMap.put("needsPosting", "Yes");
+        expectedMap.put("doesNotNeedPosting", "No");
+        expectedMap.put("courtName", LOCAL_AUTHORITY_NAME);
+        expectedMap.put("respondentLastName", "Jones");
+        expectedMap.put("representatives", List.of("Paul Blart\nStreet, Town, Postcode"));
+        expectedMap.put("caseUrlOrDocumentLink", jsonFileObject);
```

Comment:
might be too many lines, but Map.of() ?

Reply:
or does it need to be mutable?

Reply:
needs to be mutable as we will remove caseUrlOrDocumentLink before we assert the other fields